src/main/java/sirius/kernel/xml/Outcall.java

```diff
@@ -519,28 +512,77 @@ public Charset getContentEncoding() {
         }
     }
 
-    /**
-     * Sets a HTTP cookie
-     *
-     * @param name  name of the cookie
-     * @param value value of the cookie
-     */
-    public void setCookie(String name, String value) {
-        if (Strings.isFilled(name) && Strings.isFilled(value)) {
-            setRequestProperty("Cookie", name + "=" + value);
+    private void connect() throws IOException {
+        Watch watch = Watch.start();
+        try {
+            if (response != null) {
+                return;
+            }
+            if (client == null) {
+                client = clientBuilder.build();
+            }
+            if (request == null) {
+                if (postFromOutput) {
+                    requestBuilder.POST(HttpRequest.BodyPublishers.ofByteArray(out.toByteArray()));
+                }
+                request = requestBuilder.build();
+            }
+            response = client.send(request, bodyHandler);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new IOException("Thread was interrupted!");
+        } catch (IOException e) {
+            addToTimeoutBlacklist();
+            throw e;
+        } finally {
+            if (response.body() instanceof InputStream) {
```

Comment:
why don't we always use inputstreams here? we kind of rely on the counter below to count all outcalls